File: micronaut-mcp-server/src/test/java/io/micronaut/mcp/server/stateless/sync/tools/Tools.java
package io.micronaut.mcp.server.stateless.sync.tools;
/*
//tag::fakepackage[]
package example.micronaut;

//end::fakepackage[]
 */
//tag::imports[]
import io.micronaut.context.annotation.Requires;
import io.micronaut.mcp.annotations.Tool;
import jakarta.inject.Singleton;
//end::imports[]

@Requires(property = "spec.name", value = "StatelessSyncToolAnnotationTest")
//tag::clazz[]
@Singleton
class Tools {
    @Tool(description = "Evaluate a chess position using a FEN string.")
    String fenEvaluation(String fen) {
        if (fen.equals("r1bqk2r/ppp2ppp/2n5/1BbpP3/3Nn3/8/PPP2PPP/RNBQK2R w KQkq - 1 8")) {
            return "+0.27";
        }
        return "+0.0";
    }
}
//end::clazz[]

File: test-suite-docs-python/src/test/python/micronaut/mcp/docs/tools/toolarg/Tools.py
# tag::imports[]
from typing import Annotated

from io.modelcontextprotocol.common import McpTransportContext
from jakarta.inject import Singleton
from micronaut.context.annotation import Requires
from micronaut.mcp.annotations import Tool, ToolArg
# end::imports[]


@Requires(property="spec.name", value="ToolArgToolsTest")
# tag::clazz[]
@Singleton
class Tools:
    @Tool(name="fenEvaluation", description="Evaluate a chess position using a FEN string.")
    def forsyth_edwards_notation_evaluation(self,
                                            forsyth_edwards_notation: Annotated[str, ToolArg(name="fen")],
                                            ctx: McpTransportContext) -> str:
        if forsyth_edwards_notation == "r1bqk2r/ppp2ppp/2n5/1BbpP3/3Nn3/8/PPP2PPP/RNBQK2R w KQkq - 1 8":
            return "+0.12"
        return "+0.0"
# end::clazz[]
